controllers/numaresourcesoperator_controller.go

```diff
@@ -274,25 +287,48 @@ func (r *NUMAResourcesOperatorReconciler) reconcileResource(ctx context.Context,
 	return ctrl.Result{}, status.ConditionAvailable, nil
 }
 
-func (r *NUMAResourcesOperatorReconciler) syncDaemonSetsStatuses(ctx context.Context, rd client.Reader, daemonSetsInfo []nropv1.NamespacedName) ([]nropv1.NamespacedName, bool, error) {
-	dsStatuses := []nropv1.NamespacedName{}
-	for _, nname := range daemonSetsInfo {
+func (r *NUMAResourcesOperatorReconciler) syncDaemonSetsStatuses(ctx context.Context, rd client.Reader, daemonSetsInfo map[string]nropv1.NamespacedName) (map[string]nropv1.NamespacedName, bool, error) {
+	dssWithReadyStatus := map[string]nropv1.NamespacedName{}
+	for mcp, nname := range daemonSetsInfo {
 		ds := appsv1.DaemonSet{}
 		dsKey := client.ObjectKey{
 			Namespace: nname.Namespace,
 			Name:      nname.Name,
 		}
 		err := rd.Get(ctx, dsKey, &ds)
 		if err != nil {
-			return dsStatuses, false, err
+			return dssWithReadyStatus, false, err
 		}
 
 		if !isDaemonSetReady(&ds) {
-			return dsStatuses, false, nil
+			return dssWithReadyStatus, false, nil
 		}
-		dsStatuses = append(dsStatuses, nname)
+		dssWithReadyStatus[mcp] = nname
 	}
-	return dsStatuses, true, nil
+	return dssWithReadyStatus, true, nil
+}
+
+func syncNodeGroupsStatusPerMCPWithOperatorStatus(instance *nropv1.NUMAResourcesOperator, syncConfig bool) []nropv1.NodeGroupStatus {
+	ngStatuses := []nropv1.NodeGroupStatus{}
+	for _, mcp := range instance.Status.MachineConfigPools {
```

Comment:
so this logic is only valid for OpenShift and we'll add the HyperShift logic later on?

Reply:
correct, this will be handled here based on our conversations to separate the content of the OCP vs HCP:
https://github.com/openshift-kni/numaresources-operator/pull/1027